src/dbt_mcp/telemetry/logging.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+import logging
+import os
+from pathlib import Path
+
+FILE_LOGGING_ENV_VAR = "FILE_LOGGING"
+LOG_FILENAME = "dbt-mcp.log"
+
+
+def _find_repo_root() -> Path:
+    module_path = Path(__file__).resolve().parent
+    home = Path.home().resolve()
+    for candidate in [module_path, *module_path.parents]:
+        if (
+            (candidate / ".git").exists()
+            or (candidate / "pyproject.toml").exists()
+            or candidate == home
+        ):
+            return candidate
+    return module_path
+
+
+def configure_file_logging() -> None:
+    if os.environ.get(FILE_LOGGING_ENV_VAR, "").lower() != "true":
```

Comment:
Should we use `pydantic-settings` like we do for the config?

Here, we would need the value to be `True`/`true`/`TRUE` but with pydantic-settings we would also allow `t`, `on`, `1` etc...